drivers/pwm/pwm_renesas_ra8.c

```diff
@@ -0,0 +1,584 @@
+	uint16_t overflow_event;
+
+#ifdef CONFIG_PWM_CAPTURE
+	struct pwm_ra8_capture_data capture;
+#endif /* CONFIG_PWM_CAPTURE */
+};
+
+struct pwm_ra8_config {
+	const struct device *clock_dev;
+	struct clock_control_ra_subsys_cfg clock_subsys;
+	const struct pinctrl_dev_config *pincfg;
+};
+
+static uint32_t pwm_ra8_gtior_calculate(timer_cfg_t const *const p_cfg,
+					gpt_pin_level_t const stop_level)
+{
+	/* The stop level is used as both the initial level and the stop level. */
+	uint32_t gtior = R_GPT0_GTIOR_OAE_Msk | ((uint32_t)stop_level << R_GPT0_GTIOR_OADFLT_Pos) |
+			 ((uint32_t)stop_level << GPT_PRV_GTIOR_INITIAL_LEVEL_BIT);
+
+	uint32_t gtion = GPT_PRV_GTIO_LOW_COMPARE_MATCH_HIGH_CYCLE_END;
+
+	if (p_cfg->mode == TIMER_MODE_PWM) {
+		/* Use default: GTIOn is high at cycle end, then low at compare match. */
+	}
+
+#if GPT_PRV_EXTRA_FEATURES_ENABLED == GPT_CFG_OUTPUT_SUPPORT_ENABLE
+	else if (p_cfg->mode >= TIMER_MODE_TRIANGLE_WAVE_SYMMETRIC_PWM) {
+		gtion = GPT_PRV_GTIO_TOGGLE_COMPARE_MATCH;
+	}
+#endif
+	else if (p_cfg->mode == TIMER_MODE_ONE_SHOT_PULSE) {
+		gtion = GPT_PRV_GTIO_TOGGLE_COMPARE_MATCH;
+	} else {
+		/* In one-shot mode, the output pin goes high after the first compare match (one
+		 * cycle after the timer starts counting).
+		 */
+		if (stop_level == GPT_PIN_LEVEL_LOW) {
+			gtion = GPT_PRV_GTIO_HIGH_COMPARE_MATCH_LOW_CYCLE_END;
+		}
+	}
+
+	gtior |= gtion;
+
+	return gtior;
+}
+
+static int pwm_ra8_apply_gtior_config(gpt_instance_ctrl_t *const p_ctrl,
+				      timer_cfg_t const *const p_cfg)
+{
+	gpt_extended_cfg_t *p_extend = (gpt_extended_cfg_t *)p_cfg->p_extend;
+	uint32_t gtior = p_extend->gtior_setting.gtior;
+
+#if GPT_CFG_OUTPUT_SUPPORT_ENABLE
+
+	/* Check if custom GTIOR settings are provided. */
+	if (p_extend->gtior_setting.gtior == 0) {
+		/* If custom GTIOR settings are not provided, calculate GTIOR. */
+		if (p_extend->gtioca.output_enabled) {
+			uint32_t gtioca_gtior =
+				pwm_ra8_gtior_calculate(p_cfg, p_extend->gtioca.stop_level);
+
+			gtior |= gtioca_gtior << R_GPT0_GTIOR_GTIOA_Pos;
+		}
+
+		if (p_extend->gtiocb.output_enabled) {
+			uint32_t gtiocb_gtior =
+				pwm_ra8_gtior_calculate(p_cfg, p_extend->gtiocb.stop_level);
+
+			gtior |= gtiocb_gtior << R_GPT0_GTIOR_GTIOB_Pos;
+		}
+	}
+#endif
+
+#if GPT_PRV_EXTRA_FEATURES_ENABLED == GPT_CFG_OUTPUT_SUPPORT_ENABLE
+	gpt_extended_pwm_cfg_t const *p_pwm_cfg = p_extend->p_pwm_cfg;
+
+	if (NULL != p_pwm_cfg) {
+		/* Check if custom GTIOR settings are provided. */
+		if (0 == p_extend->gtior_setting.gtior) {
+			/* If custom GTIOR settings are not provided, set gtioca_disable_settings
+			 * and gtiocb_disable_settings.
+			 */
+			gtior |= (uint32_t)(p_pwm_cfg->gtioca_disable_setting
+					    << R_GPT0_GTIOR_OADF_Pos);
+			gtior |= (uint32_t)(p_pwm_cfg->gtiocb_disable_setting
+					    << R_GPT0_GTIOR_OBDF_Pos);
+		}
+	}
+#endif
+
+	/* Check if custom GTIOR settings are provided. */
+	if (0 == p_extend->gtior_setting.gtior) {
```

Comment:
`if (p_extend->gtior_setting.gtior == 0) {`

Reply:
I updated it, thank you